feature/ec2/imds/request_middleware.go

```diff
@@ -224,10 +242,11 @@ func (m *operationTimeout) HandleInitialize(
 ) (
 	output middleware.InitializeOutput, metadata middleware.Metadata, err error,
 ) {
-	var cancelFn func()
-
-	ctx, cancelFn = context.WithTimeout(ctx, m.Timeout)
-	defer cancelFn()
+	if _, ok := ctx.Deadline(); !ok {
+		var cancelFn func()
+		ctx, cancelFn = context.WithTimeout(ctx, m.DefaultTimeout)
```

Comment:
Having the `DefaultTimeout` member on `operationTimeout` middleware is handy if we want to change the timeout depending on the operation. I'm not aware of a need for this at the moment. Since `operationTimeout` is not exported, having the member is preferred so we can reuse the operationTimeout middleware later. 

There is a test that uses a custom timeout, but I think this test needs to be tweaked a little bit to more closely align with the intention behind it. e.g. if there was a context with timeout use it, if there wasn't use a default. This test is relying on the behavior of a canceled context, and the configured nanosecond timeout input.

https://github.com/aws/aws-sdk-go-v2/blob/dccd10a26f9091d3123bd1fc63ef572dd60fb218/feature/ec2/imds/request_middleware_test.go#L128-L152